packages/builder/src/pages/builder/app/[application]/design/[screenId]/_components/AppPreview.svelte

```diff
@@ -125,7 +125,7 @@
       // Catch any app errors
       loading = false
       error = event.error || "An unknown error occurred"
-    } else if (type === "select-component" && data.id) {
+    } else if (type === "select-component" && "id" in data) {
```

Comment:
Since we aren't sending up nulls any more, we should probably revert this too to ensure we don't ever store a null unintentionally.